### PR TITLE
Bundle collection to html by default

### PR DESCRIPTION
**Pull request by thangngoc89**
Continue from #98 
ref #92 #96 

This PR will bundle collection to html by  default.

We can disable this behavior on page specific by passing an option `collection: false` to it's head

Example:
```
---
layout: Post
title: Some page you don't want to bundle collection by default
collection: false
---
```

TODO:

- [x] Add collection to bundle
- [x] Don't fetch collection when it is already in store
- [x] Allow to disable this behaviour
- [ ] Fix tests

### CODE REVIEW

**Comment by thangngoc89** [comment]
@MoOx could you help me with these test ?
I'm not so familiar with ava.

**Comment by MoOx** [comment]
Can you rebase/squash on master and drop the changes in src/PageContainer/__tests__/component.js? (I fixed this)

File: statinamic [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules/statinamic
```
Comment (MoOx) [inline statinamic]: what is this file?

**Comment by MoOx** [comment]
For ava you should be able to debug using `ava --verbose the/file/to/run.js`.

**Comment by MoOx** [comment]
FYI, I think for now, I will just completely remove collection.json, so we can even drop this option.
I don't really see any use case for having a json if we already have the collection in any entry points.
Make sense?

File: statinamic (hunk 1 above)
Comment (thangngoc89) [inline statinamic]: oops. my bad

**Comment by thangngoc89** [comment]
Yes. Since we only fetch collection.json once. And we did bundle it inside every HTML file. We can just remove it and make code cleaner

**Comment by MoOx** [comment]
I am on it. This require serving prerendered pages during development (which is not the case atm, see #23)

**Comment by thangngoc89** [comment]
Rebased

**Comment by thangngoc89** [comment]
@MoOx should I remove the option to not bundle collection in HTML ?

**Comment by MoOx** [comment]
Probably. I think we should not touch this PR until I end my current refactoring. I will notify you when it's done.

**Comment by thangngoc89** [comment]
Close in favor of #113

**Comment by MoOx** [comment]
Note that the PR #113 description contained enough ("close #113") to automatically close this issue when merged into master :)

**Comment by thangngoc89** [comment]
Didn't know about that :+1:

**Pull request closed without merging**